### PR TITLE
Improve processing api

### PR DESCRIPTION
**Pull request by Aadjou**
Scope & Features:
- This PR implements the changes to the improved processing API design.
- Affected / changed / created 3dio methods: `io3d.light.bake`, `io3d.light.bakeLoRes`, `io3d.light.bakeHiRes`, `io3d.modify.origami`, `io3d.modify.consolidateFaceSides`, `io3d.storage.exportBlend`, `io3d.storage.export3ds`, `io3d.storage.exportFbx`, `io3d.storage.exportDae`, `io3d.storage.exportDxf`, `io3d.storage.exportObj`

Design:
- [Processing API design](https://github.com/archilogic-com/services/issues/279) (internal)
- Updated API documentation draft for [modify](https://3d.io/branch/processing-api-documentation/docs/api/1/modify.html), [light baking](https://3d.io/branch/processing-api-documentation/docs/api/1/lightmap-baking.html), [model export](https://3d.io/branch/processing-api-documentation/docs/api/1/model-export.html)

How to use:
- Please consult the documentation draft on how to use the new api
- Set your config files to use the testing web-server if you want to test the changes.

Notes:
- If there are any questions, please do not hesitate to ask me

### CODE REVIEW

File: src/light/bake.js [hunk 1]
```diff
@@ -3,34 +3,39 @@ import callServices from '../utils/services/call.js'
...
-      inputFileKey: storageId,
-      options: {
-        inputAssetKeys: assetStorageIds,
-        sunDirection: sunDirection,
-        cacheKey: cacheKey
+    var bakeParams = {
+      method: 'bake'.concat('.', stage),
+      params: {
+        inputFileKey: storageId,
+        //inputAssetKeys: assetStorageIds,
```
Comment (AVGP) [inline src/light/bake.js]: did we remove this option or why is it commented out?

File: src/light/bake.js [hunk 2]
```diff
@@ -3,34 +3,39 @@ import callServices from '../utils/services/call.js'
...
-  // API
-  options = options || {}
-  var sunDirection = sunDirection || [
-    0.7487416646324341,
-    -0.47789104947352223,
-    -0.45935396425474223
-  ]
+    // Optional bake parameters for API call
+    var bakeSettings = { sunDirection: options.sunDirection || [ 0.75, -0.48, -0.46 ] }
+    if (options.lightMapCount) { bakeSettings.lightMapCount = options.lightMapCount }
```
Comment (tomas-polach) [inline src/light/bake.js]: this has to be either (prefered option in one line without semicolon must not have curly brackets)
```
if (options.lightMapCount) bakeSettings.lightMapCount = options.lightMapCount
```
or (without semicolon with curly brackets: use multiple lines)
```
if (options.lightMapCount) {
  bakeSettings.lightMapCount = options.lightMapCount
}
```
or (rather not common: in one line with curly brackets: requires semicolon)
```
if (options.lightMapCount) { bakeSettings.lightMapCount = options.lightMapCount; }
```

File: src/light/bake.js [hunk 3]
```diff
@@ -3,34 +3,39 @@ import callServices from '../utils/services/call.js'
...
+    if (options.samples) { bakeSettings.samples = options.samples }
```
Comment (tomas-polach) [inline src/light/bake.js]: see comment for line 13

File: src/light/bake.js (hunk 1 above)
Comment (Aadjou) [reply src/light/bake.js]: it's currently an empty array hardcoded. i can remove it

File: src/light/bake.js (hunk 2 above)
Comment (Aadjou) [reply src/light/bake.js]: thanks!

File: src/light/bake.js (hunk 1 above)
Comment (AVGP) [reply src/light/bake.js]: nah, was just wondering

**Comment by Aadjou** [comment]
Thanks for the review!

**Pull request merged**